cli/src/pcluster/validators/networking_validators.py

```diff
@@ -75,3 +75,51 @@ def _validate(self, queues_subnets):
         subnet_ids = {tuple(set(queue_subnets)) for queue_subnets in queues_subnets}
         if len(subnet_ids) > 1:
             self._add_failure("The SubnetId used for all of the queues should be the same.", FailureLevel.ERROR)
+
+
+class LambdaFunctionsVpcConfigValidator(Validator):
+    """Validator of Pcluster Lambda functions' VPC configuration."""
+
+    def _validate(self, security_group_ids: List[str], subnet_ids: List[str]):
+        existing_security_groups = AWSApi.instance().ec2.describe_security_groups(security_group_ids)
+        existing_subnets = AWSApi.instance().ec2.describe_subnets(subnet_ids)
+
+        self._validate_all_security_groups_exist(existing_security_groups, security_group_ids)
+        self._validate_all_subnets_exist(existing_subnets, subnet_ids)
+        self._validate_all_resources_belong_to_the_same_vpc(existing_security_groups, existing_subnets)
+
+    def _validate_all_resources_belong_to_the_same_vpc(self, existing_security_groups, existing_subnets):
+        group_vpc_ids = {group["VpcId"] for group in existing_security_groups}
+        subnet_vpc_ids = {subnet["VpcId"] for subnet in existing_subnets}
+        if len(group_vpc_ids) > 1:
+            self._add_failure(
+                "The security groups associated to the Lambda are required to be in the same VPC.", FailureLevel.ERROR
```

Comment:
minor: you could avoid duplication creating a format string to be used in every case:
```python
message = "The {} associated to the Lambda are required to be in the same VPC."
```
then you can use:
```python
message.format("security groups")
message.format("subnets")
message.format("security groups and subnets")
```
in the different cases.

Reply:
I don't have strong opinions on this, but imho it wouldn't really make the code clearer/more explicit.